secboot/secboot.go

```diff
@@ -28,11 +28,46 @@ import (
 	"github.com/snapcore/snapd/asserts"
 )
 
+const (
+	RecoveryBootloaderAsset = iota
+	RecoveryBootloaderKernel
+	BootloaderAsset
+	BootloaderKernel
+)
+
+// EFIImage represents an EFI binary (or snap file containing an EFI
+// binary) that is part of a load sequence.
+type EFIImage struct {
+	// Type specified the role and origin of this image
+	Type int
```

Comment:
I don't think we should conflate the the role and type kind way. Also we shouldn't assume that kernels are in snaps and assets outside of snaps.

My preference is also still for this kind of type to live directly in bootloader as BootFile or something like that.